src/main/java/com/azure/cosmos/kafka/connect/source/CosmosDBSourceConnector.java

```diff
@@ -50,8 +50,9 @@ public void start(Map<String, String> props) {
             // Equally distribute workers by assigning workers to containers in round robin fashion.
             Map<String, String> taskProps = config.originalsStrings();
             taskProps.put(CosmosDBSourceConfig.COSMOS_ASSIGNED_CONTAINER_CONF,
-                          containerList[i % containerList.length]);
-            taskProps.put(CosmosDBSourceConfig.COSMOS_WORKER_NAME_CONF, config.getWorkerName() + i);
+                          containerList.get(i % containerList.size()));

```

Comment:
Curious if this needs to be unique for entire connector since now lease-prefix is required to be consistent across multiple workers?

Reply:
Not sure what you mean. Each worker/task will still get it's own assigned container and if there are multiple tasks but a single container in the topic-map, the assigned container should be the same for all tasks. This should mean the lease prefix is the same too.

Reply:
that makes sense